### PR TITLE
Gambling machines now payout again

### PR DESCRIPTION
**Pull request by Ty-Omaha**
~~Reverts https://github.com/ParadiseSS13/Paradise/pull/8182 - and the subsequent https://github.com/ParadiseSS13/Paradise/pull/8222~~

~~As it is right now - there is absolutely no reliable way to earn money.~~ Gambling machines sit idle, ignored in the bar because of the realism slap.

~~Things that can be bought with money are novel, and there is really no downside in not restricting those items.~~

~~If an attempt at an economy or making money "useful" wants to be done, it should be done all together instead of removing the only reliable way to make money while keeping all the other aspects and doing the rest later.~~

### **Edit: Compared to how it used to be, it is now to 2x harder to win something and 3x harder for the 1 mil & 100k jackpot**
_(*unless my math is wrong which wouldn't be surprising and it's really like 1.5x harder. correct me if that's the case)_

While it won't be a "money printer" again now, it certainly won't be a scam like the current iteration.

:cl:
tweak: Gambling machines now payout again.
/:cl:

### CODE REVIEW

**Comment by Birdtalon** [comment]
Just give everyone more money to begin with. Gambling machines being able to just infinitely make money is silly and removes anything special as regards "getting lucky."

**Comment by Ty-Omaha** [comment]
Nobody bothers to use the gambling machines to try and "get lucky" because the odds are astronomically against them. Gambling machines gave people something to mess around with. Having a clown slipping people on a space station is also silly - as with a lot of aspects of SS13.

**Comment by LightFire53** [comment]
This would be something that should be tweaked, not reverted. I can't say I want to go back to having 3 or 4 millionaires a shift again. Change the numbers, not revert.

**Comment by Regen1** [comment]
>Mess around with

Literally pressing one button, waiting for RnG hits and you get an arbitrary currency not really used anywhere? Yeah nah.

Make a new currency called "regenbux", you'll get them from a machine that has a big red button, it has a small percentage got give a random value of regenbux, that'll keep 'em busy for ages

**Comment by Ty-Omaha** [comment]
>Change the numbers, not revert.
I'll ask around about that, seems like a good idea.

> Literally pressing one button, waiting for RnG hits and you get an arbitrary currency not really used anywhere? Yeah nah.
Yep, people "mess around" with stuff like that. Cookie clicker is a good example of that.

**Comment by taukausanake** [comment]
Make a new slot machine retrofitted from a peaceborg, kind of like how the janicart is a dead janiborg. Make the slot dispense cookies when you win instead of more money.
Additionally you can make it emaggable to dispense a chaos cookie (or are those just donuts? Should make a cookie too). That would be cool

**Comment by The-Driker** [comment]
slots

I did 45 spins in a row and wasted all of my money to test how rigged it is now. I won a total of 20 credits, which was two 10 credit wins. 

The nerf literally made this not worth using anymore and worthless mechanics that noone wants to use, need to not be in the game. Period. 

Hopefully the numbers you adjust it to make it worth playing with.

**Comment by Anasari** [comment]
I don't believe the slot machine is a good idea in the first place. 

In any form, it either rewards or punish the player for left clicking a single button in a single location non-stop.

There's no skills, no meaningful interaction, no depth in such a mechanic. I'd rather people played blackjack against a machine or did poker to earn their money. There is a tangible "Skill factor" with those game.

However a payout rate tweak to make it closer to real life, with roughly -1 to -2 credits loss per bet, and way more often small payout would be nice.

You also have to consider that money is practically rendered useless both practically and symbolically by the old slot machine. Because money is so easy to come by on the slot machine, the rest of the station's money, the secure briefcase, etc., are useless due to how easy it is to acquire money.

Making sure money can't be earned easily by left clicking on a machine restores some of the roleplay value in credits.

**Comment by Tayyyyyyy** [comment]
If there's no downside in restricting the items, make them free, or lower their cost, or give everybody more starting money. It doesn't make sense to buff gambling machines again. It's an unnecessary middleman.

Slot machines are not supposed to be reliable money makers. If you want people to reliable earn money, maybe give them money every five minutes or something. Reverting the slot machines PR is a knee-jerk reaction that doesn't improve the game.

**Comment by Ty-Omaha** [comment]
>I don't believe the slot machine is a good idea in the first place.

It's the only way to make money for items (instruments, snap pops, and other fun misc things.) apart from going around begging people for money.

>In any form, it either rewards or punish the player for left clicking a single button in a single location non-stop. 

As it is now, it is "click me to have your money completely drained".  It's not a fun mechanic for anybody. 
 No one even bothers to use it anymore - it's the same as outright removing them.
i.e.
>_"The nerf literally made this not worth using anymore and worthless mechanics that noone wants to use, need to not be in the game. Period." -Driker_ 

.
>You also have to consider that money is practically rendered useless both practically and symbolically by the old slot machine.

Money is /still/ practically and symbolically useless. What can it buy? Fun little nick nacs and vending machine food.

>Making sure money can't be earned easily by left clicking on a machine restores some of the roleplay value in credits.

I haven't seen anyone roleplay with credits _more_ since the nerf, and the people who actually  _would_ roleplay with them would still do so even if there is a "money printing machine".
I've seen much more roleplay and people interacting with eachother from reacting to 10k winnings.


>Reverting the slot machines PR is a knee-jerk reaction that doesn't improve the game.

Having a thing where nobody bothers to touch it because the odds are astronomically against them doesn't improve the game.

I also think it does improve the game; as traders will be able to get money again and buy a ton of useless nick nacks to sell/trade. Additionally, people won't be shafted for useless but "fun" items that they might want and people will have something to mess around with if they're bored. 

If someone wants to make another way to get money then by all means; but so far nobody has.

**Comment by Tayyyyyyy** [comment]
Slot machines are high risk, high reward. They should not be low risk, high reward. If you pull a slot machine 45 times, you should be expected to lose money on average. However, you can get lucky and win a huge amount of money when you do win.

> If someone wants to make another way to reliably get money then by all means; but so far nobody has.

Your main reason for this PR is that people can't get those expensive items anymore, which I agree is a bad thing. We're just suggesting other ways of fixing this problem without bringing back slot machines.

**Comment by Ty-Omaha** [comment]
I went ahead and edited the numbers a bit, hopefully people find this as a fair middleground.

It won't be a "money printer" like it used to, but it certainly wont be a waste of time like it is now.

**Comment by Tayyyyyyy** [comment]
Do you have any numbers on what the expected value of pulling the slot machine is? If the expected value is positive, then there's no difference from what it was before. It just takes more tries to get money, which doesn't improve the situation for anyone; you might as well do a full revert or just give people 1 million dollars every time they press the button. Slot machines are supposed to be a scam! That's how they make money. It absolutely makes no sense for a slot machine that gives you money. Here's why [slot machines still exist](http://triblive.com/x/pittsburghtrib/ae/gambling/s_777687.html) even though 100% rational people would never touch them. And [a paper on the psychology of slot machines](https://link.springer.com/content/pdf/10.1007%2Fs10899-009-9156-6.pdf) if that's what you're into.

Here's the file you can edit to adjust the prices on the store items:
https://github.com/ParadiseSS13/Paradise/blob/18f4aa18b98fcf8503f7173d3e294d581aa2acff/code/modules/store/items.dm
I think it would be a much better solution to just tone down those numbers. Maybe divide them all by 10 or something. Why add more pulls when you can just make the store items more affordable?

**Comment by Ty-Omaha** [comment]
>which doesn't improve the situation for anyone;

Nobody even uses slot machines anymore. At least people will use them now.

> Slot machines are supposed to be a scam! That's how they make money. It absolutely makes no sense for a slot machine that gives you money. Here's why slot machines still exist even though 100% rational people would never touch them. And a paper on the psychology of slot machines if that's what you're into.

The difference is, this is a game with rounds that last roughly 2 hours, and in real life people still **use** slot machines.
Ever since the nerf nobody cares about them because it's chances are ridiculous; as Driker pointed out.


>Do you have any numbers on what the expected value of pulling the slot machine is?

Not that good with math so I tested it out and of 100 tries, I won:
`10 - 1000 - 10 - 50 - 1000 -10 - 10 - 50 - 10 - 50`

Had I not gotten a 1000 winning I would have been in the negatives.

**Comment by Tayyyyyyy** [comment]
I don't get why slot machines have to be used, though. Their main purpose is to get money to buy items from the item store, and you can just reduce the price on those items. If the expected value is positive, there's really no point to having a "middle ground". You're just adding a little bit of inconvenience to the process. I'm saying you should get rid of that inconvenience and just cut the prices on the items. Especially because right now we have no idea where this puts slot machines unless Ansari swoops in and saves us with his math skills :p

**Comment by outbools** [comment]
>Having a clown slipping people on a space station is also silly - as with a lot of aspects of SS13. 

wow we back again with false analogies, making mistakes and justifying them by older mistakes?

nice slippery slope, fellow coder

**Comment by Kyep** [comment]
I agree with Ansari. Slot machines should be removed entirely.
Reasons:
- Spam-clicking on an RNG machine is not deep, interesting, or deserving of reward.
- Rewarding spam-clicking with credits makes credits useless, because you can get a huge amount by just clicking a machine many times.
- The announcements that the machines generate are annoying, and, prior to the nerf, far, far too common. One in a million chance? More like one in a hundred, IE: effectively guaranteed if you spam it for a few minutes.

If you want to make it possible to win credits by playing games... require the player to beat the chess computer. Or engage in some other game of skill. Something that requires thought, isn't spammable, and is NOT announced to crew when you win.

**Comment by Tayyyyyyy** [comment]
> Or engage in some other game of skill.

[Video poker](https://en.wikipedia.org/wiki/Video_poker) is another casino game that would work. It's a game that you can adjust the payouts so a player who plays well will win money on average, but the house will still win vs unskilled players. Obviously this is a bit trickier to implement than slot machines...

**Comment by LightFire53** [comment]
The point of reducing chances to win would reduce the amount of announcements, @Kyep . Nor do I think we need to remove them. It's something people can do to occupy their time, even if it is more used than the holodeck.

**Comment by Dumbdumn5** [comment]
The previous PR being made with the intention of creating a 'house always wins' scenario was something that brought a little too much unfairness into the whole economy which barely has much of a use right now as it stands, outside of the cargo merch computer and whenever the bartender charges people and somehow doesn't get murdered for it. As such, this PR's function as a partial revert still makes it possible to actually gain something from playing the slots, meaning you can 'win' them, but with a low enough chance that you'll have to risk putting yourself in the hole or going bankrupt before you can actually get anything good from it.

Due to these winrate changes, we feel that this will make slot machines a little more enticing to play without making them completely and utterly pointless or functional cash sinks that nobody would use. 

tl;dr :+1: due to the harder and less forgiving winrate but without making it functionally impossible to actually win anything back.

**Comment by tigercat2000** [comment]
^
